src/python/pants/scm/git.py

```diff
@@ -571,6 +573,7 @@ def _read_tree(self, path):
     object_type, tree_data = self._read_object_from_repo(rev=self.rev, relpath=path)
     assert object_type == b'tree'
     # The tree data here is (mode ' ' filename \0 20-byte-sha)*
+    tree_data = [bytes([b]) for b in tree_data]
```

Comment:
@stuhood This commit is what code looks like when having _read_object_from_repo return a bytestring instead of `List[byte char]`. 

Let me know if you think I should revert this to before. It looks like the function is private to the repo, so I'm in favor of reverting to before as it results in 3 less lines of duplicated code.

Reply:
Between:
```
    blob = self._cat_file_process.stdout.read(int(object_len))
    blob = [bytes([b]) for b in bytes(blob)]
```
...and:
```
    blob = bytes(self._cat_file_process.stdout.read(int(object_len)))
```
... definitely the latter, right?

Reply:
From what I can tell, most/all of those `tree_data = [bytes([b]) for b in tree_data]` transforms are because we're calling `join` on bytes, which doesn't seem necessary.